### ISSUE TITLE
Global/Local prefixes

### ISSUE THREAD

**Issue by vzaliva** [ProofGeneral/PG#182]
I have noticed that `Global` and `Local` prefixes are treated differently.
See attached screenshot.

Reference to Coq documentation regarding these keywords: https://coq.inria.fr/distrib/current/refman/Reference-Manual008.html#hevea_default470

![screenshot from 2017-05-02 19-13-25](https://cloud.githubusercontent.com/assets/86581/25646268/a08cb670-2f6b-11e7-97fb-4c5734a67dad.png)

**Comment by cpitclaudel** [comment]
It's not that, IIRC - since they don't apply to everything we don't have local and global as keywords - instead, we add them for each individual option, so to fix this problem we just need to add particular option.
Whether that's the right approach is debatable, of course (compared to just highlighting Local)

**Comment by Matafou** [comment]
No strong opinion about that. Since it is quite difficult to track all the
command that accept Local/Global and Set/Unset we should maybe swich to a
simpler approach indeed.

P.

2017-05-03 6:20 GMT+02:00 Clément Pit-Claudel <notifications@github.com>:

**Comment by vzaliva** [comment]
I looked a bit at Coq sources trying to get the list of vernacular supporting locality prefixes (Local/Global) and here is what I found. Hopefully, this will help to classify them all correctly in PG:

```
(* Vernaculars that take a locality flag *)
let check_vernac_supports_locality c l =
  match l, c with
  | None, _ -> ()
  | Some _, (
      VernacOpenCloseScope _
    | VernacSyntaxExtension _ | VernacInfix _ | VernacNotation _
    | VernacDefinition _ | VernacFixpoint _ | VernacCoFixpoint _
    | VernacAssumption _ | VernacStartTheoremProof _
    | VernacCoercion _ | VernacIdentityCoercion _
    | VernacInstance _ | VernacDeclareInstances _
    | VernacDeclareMLModule _
    | VernacCreateHintDb _ | VernacRemoveHints _ | VernacHints _
    | VernacSyntacticDefinition _
    | VernacArgumentsScope _ | VernacDeclareImplicits _ | VernacArguments _
    | VernacGeneralizable _
    | VernacSetOpacity _ | VernacSetStrategy _
    | VernacSetOption _ | VernacSetAppendOption _ | VernacUnsetOption _
    | VernacDeclareReduction _
    | VernacExtend _ 
    | VernacInductive _) -> ()
  | Some _, _ -> CErrors.error "This command does not support Locality"

```

**Comment by Matafou** [comment]
Thanks, that is useful. It will change over time though.
P.


2017-05-03 17:22 GMT+02:00 Vadim Zaliva <notifications@github.com>:

> I looked a bit at Coq sources trying to get the list of vernacular
> supporting locality prefixes (Local/Global) and here is what I found.
> Hopefully, this will help to classify them all correctly in PG:
>
>

**Comment by cpitclaudel** [comment]
Indeed, thanks. Re changing over time, there's hope that the new manual format will provide this kind of data in machine readable format, so with a bit of luck we'll be able to auto generate these lists in the future

**Comment by erikmd** [comment]
For the record, this issue seems related to issue #73
